sql/structured.go

```diff
@@ -128,40 +128,32 @@ func validateName(name, typ string) error {
 }
 
 // containsColumnID returns true if the index descriptor contains the specified
-// column ID.
+// column ID either in its explicit column IDs or the implicit "extra" column
+// IDs.
 func (desc *IndexDescriptor) containsColumnID(colID ColumnID) bool {
 	for _, id := range desc.ColumnIDs {
 		if id == colID {
 			return true
 		}
 	}
+	for _, id := range desc.ExtraColumnIDs {
+		if id == colID {
+			return true
+		}
+	}
 	return false
 }
 
 // fullColumnIDs returns the index column IDs including any implicit column IDs
 // for non-unique indexes.
-func (desc *IndexDescriptor) fullColumnIDs(tableDesc *TableDescriptor) []ColumnID {
-	// Verify that the index is associated with the table descriptor. Note that
-	// we're using pointer comparisons here!
-	if desc != &tableDesc.PrimaryIndex {
-		found := false
-		for i := range tableDesc.Indexes {
-			if desc == &tableDesc.Indexes[i] {
-				found = true
-				break
-			}
-		}
-		if !found {
-			panic(fmt.Sprintf("unable to find index \"%s\" in table: %+v", desc.Name, tableDesc))
-		}
-	}
-
+func (desc *IndexDescriptor) fullColumnIDs() []ColumnID {
```

Comment:
rather than adding `ExtraColumnIDs` to the descriptor, could we pass the table descriptor (or the primary index descriptor) to this method and have it compute them? we're already going to avoid calling this method repeatedly - the extra cost may not be so bad.